src/PreparedWrites.cc

```diff
@@ -0,0 +1,521 @@
+ *      This is primarily used by the multiWrite RPC in MasterService.
+ */
+PreparedOp::PreparedOp(WireFormat::TxPrepare::OpType type,
+                       uint64_t clientId,
+                       uint64_t rpcId,
+                       uint32_t participantCount,
+                       WireFormat::TxParticipant* participants,
+                       uint64_t tableId,
+                       uint64_t version,
+                       uint32_t timestamp,
+                       Buffer& keysAndValueBuffer,
+                       uint32_t startDataOffset,
+                       uint32_t length)
+    : header(type, clientId, rpcId, participantCount)
+    , participants(participants)
+    , object(tableId, version, timestamp, keysAndValueBuffer,
+             startDataOffset, length)
+{
+}
+
+/**
+ * Construct a PreparedOp in preparation for storing it in the log.
+ * Use this constructor when the data for an object is contiguous and
+ * the header information is  available in individual pieces. This is
+ * primarily used by unit tests to construct objects with just a
+ * single key. The main function that will be invoked after a call
+ * to this constructor is assembleForLog.
+ *
+ * \param type
+ *      Type of the staged operation.
+ * \param clientId
+ *      leaseId given for this linearizable RPC.
+ * \param rpcId
+ *      rpcId given for this linearizable RPC.
+ * \param participantCount
+ *      Number of objects participating current transaction.
+ * \param participants
+ *      Pointer to the array of #WireFormat::TxParticipant which contains
+ *      all information of objects participating transaction.
+ * \param key
+ *      Primary key for this object
+ * \param value
+ *      Pointer to a single contiguous piece of memory that comprises this
+ *      object's value.
+ * \param valueLength
+ *      Length of the value portion in bytes.
+ * \param version
+ *      Version number of this object, which is used to disambiguate
+ *      different incarnations of objects with the same key.
+ * \param timestamp
+ *      The creation time of this object, as returned by the WallTime
+ *      module. Used primarily by the cleaner to order live objects and
+ *      improve future cleaning performance.
+ * \param buffer
+ *      A buffer that can be used temporarily to store the keys and value
+ *      for the object. Its lifetime must cover the lifetime of this Object.
+ * \param [out] length
+ *      Total length of keysAndValue
+ */
+PreparedOp::PreparedOp(WireFormat::TxPrepare::OpType type,
+                       uint64_t clientId,
+                       uint64_t rpcId,
+                       uint32_t participantCount,
+                       WireFormat::TxParticipant* participants,
+                       Key& key,
+                       const void* value,
+                       uint32_t valueLength,
+                       uint64_t version,
+                       uint32_t timestamp,
+                       Buffer& buffer,
```

Comment:
Use a pointer, not a reference (references are deprecated in RAMCloud, uunless there's a good reason for them).